lib/rover/vector.rb

```diff
@@ -22,6 +30,18 @@ def initialize(data)
       raise ArgumentError, "Bad size: #{@data.shape}" unless @data.ndim == 1
     end
 
+    def to(type)
+      numo_type = TYPE_CAST_MAPPING[type]
+
+      if numo_type == Numo::DFloat && @data.class == Numo::RObject
+        Vector.new(@data.to_a.map { |item| item.to_f })
```

Comment:
Think we can make this `item.nil? ? Float::NAN : item.to_f` to solve the problem you mentioned.